presto-spi/src/main/java/com/facebook/presto/spi/type/SmallintType.java

```diff
@@ -93,7 +94,8 @@ public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int
     @Override
     public long hash(Block block, int position)
     {
-        return block.getShort(position, 0);
+        // Important: this implementation must match SmallintOperators.hash()
+        return XxHash64.hash(block.getShort(position, 0));
```

Comment:
The single-step murmur for long shifts everything by 33 bits, and the int version shifts by 16 bits. I don't know of a version that works for shorts, so I took the easy route.